readme.md

```diff
@@ -1,12 +1,94 @@
+    res.end(JSON.stringify(result));
+  });
+});
+```
+
+## How to install
+
+In your existing project add `mongodb-legacy` to your `package.json` with the following command.
+
+```sh
+npm install mongodb-legacy
 ```
 
-We are soon shipping our `'mongodb'` with promise only APIs, but this package can help those who have difficulty adopting that change by continuing to offer our API in it's existing form a combination of callback and promise support. We hope that this module will require only the change of the import string from `'mongodb'` to `'mongodb-legacy'` along with adding `'mongodb-legacy'` to your `package.json`. Our intent is to ensure that the existing APIs offer precisely the same behavior as before, the logic for handling callback or promise been moved to these light wrappers. Please let us know if you encounter any differences if you have need of this package.
+### Versioning
+
+There is no one size fits all solution to how your project and policies may require pulling in dependencies, below is one recommended path.
+
+**We recommend you keep `mongodb` in your project's direct dependencies**.
+The package has a direct dependency on the `mongodb` package of `^4.10.0`.
+So when you're ready to upgrade to `v4.10+` and want to start working with this wrapper package keeping `mongodb` in your project's dependencies will allow you to control upgrades explicitly as needed.
+Without `mongodb` specified in your project's dependencies it will be up to `npm` and your `package-lock.json` file updating policy to control when new driver versions are pulled into your project.
+
+The next major release of the driver (v5) will drop support for callbacks.
+This package will also have a major release at that time to update the peerDependency requirement to `v5+`.
+Users can expect to be able to upgrade to `v5` adopting the changes and features shipped in that version while using this module to maintain any callback code they still need to work on migrating.
+
+## API
+
+The API is inherited from the driver so you should be able to peruse the API docs
+
+- [Found here](https://mongodb.github.io/node-mongodb-native/Next/)
+
+The wrappers are implemented as subclasses of each of our driver's classes that you may use building your app that just handles the optional callback behavior. This means any new APIs added to the driver should be automatically be pulled in as long as the updated driver is installed.
+
+Take this hypothetical example:
+
+```ts
+// Just add '-legacy' to my mongodb import
+import { MongoClient } from 'mongodb-legacy';
+const client = new MongoClient();
+const db = client.db();
+
+const collection = db.collection('pets');
+// Returns an instance of LegacyFindCursor which is a subclass of FindCursor
+const dogCursor = collection.find({ kind: 'dog' });
+// Using .next with callbacks still works!
+dogCursor.next((error, dog) => {
+  if (error) return handling(error);
+  console.log(dog);
+});
+// Brand new api that pets all dogs!
```

Comment:
call out that the new API will not have callback support here and possibly in the paragraph below

Reply:
called out!